enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,276 @@
+
+## Motivation
+
+Currently, all addons agents should be deployed on the managed cluster, it is mandatory to expose the hub cluster to 
+the managed cluster even if the managed cluster resides in an insecure environment.
+
+The proposal [hosted-deploy-mode](./../33-hosted-deploy-mode/README.md) introduced a new deploy mode `Hosted` to deploy 
+klusterlet outside of the managed cluster, but keeps the addons agents remaining on the managed cluster, this proposal 
+makes the addon framework support running addons agents outside of the managed cluster, which is defined as `Hosted` 
+mode deploy for addons agents.
+
+### Goals
+
+- Make addon-framework deploy addons agents outside of the managed cluster.
+- Define how to deploy an addon in hosted mode
+
+### Non-Goals
+
+- Implementation details of addon
+- Support deploying all addons in hosted mode. Not all addons can run in hosted mode(ex: an addon agent 
+needs to analyze some managed cluster node related information, which can only be collected in the managed cluster and 
+cannot be exposed outside the managed cluster).
+
+## Proposal
+
+We propose to define an annotation of the `ManagedClusterAddon` for addon deployers to indicate that the addon is 
+deployed in hosted mode and where agents deployments will be deployed. And provide a label for addon developers to 
+specify which manifests need to be deployed outside the managed cluster.
+
+### User Stories
+
+#### Story 1
+
+As a managed cluster administrator, I hope as few as possible workloads running in my cluster while the addon 
+functionality is not affected.
+
+#### Story 2
+
+As a hub cluster administrator, on the premise that the addon functionality is not affected, I do not want the managed 
+cluster to talk to the hub cluster directly.
+
+### Terms
+
+- `Hosted mode`: deploy addons agents deployments outside the managed cluster.
+- `Hosting cluster`: The cluster where the addons agents deployments runn. A Hosting cluster is a managed cluster of 
+the hub, which is to host the addon agent to manage another managed cluster's addon.
+- `Addon deployer`: A person or a component responsible for deploying an addon.
+
+### Constraints
+
+There are several constraints:
+- The managed cluster must be exposed to the hosting cluster
+- The managed cluster(Klusterlet) must be imported to hub in `Hosted` mode
+- The hosting cluster must be a managed cluster of the hub
+- The hosting cluster of the addon must be the same hosting cluster of the klusterlet
+- After the addon is deployed, the deployment mode cannot be switched
+
+## Design Details
+
+### For addon developers
+
+We propose to add a label `addon.open-cluster-management.io/hosted-manifest-location` for the addon agent manifests to 
+mark which manifests should be deployed where in Hosted mode:
+- No matter what the label's value is, all manifests will be deployed on the managed cluster in Default mode.
+
```

Comment:
The following content is related with PR: https://github.com/open-cluster-management-io/registration-operator/pull/253  @elgnay @zhujian7 @qiujian16 

---

Developers should at least add a namespace resource with the label `"addon.open-cluster-management.io/namespace": "true"` for addons' hosted mode manifests.

Because manifests in hosted mode could be installed in any namespace specified by `{ManagedClusterAddon.Spec.InstallNamespace}` which may not be created or synced with image pull secret.

Here is an example:
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: {{ .Release.Namespace }}
  labels:
    addon.open-cluster-management.io/namespace: true

Reply:
Developers are required to ensure the existence of the addon agent namespace. if there is the label `"addon.open-cluster-management.io/namespace": "true"` on the namespace, the addon-framework will copy the image pull secret to the namespace, otherwise, developers need to provide the image pull secret themselves.

Reply:
@zhujian7  Agree， clear and simple!

Reply:
Maybe we can addon some context before?

---

If manifests will be deployed on the hosting cluster in Hosted mode,  then Developers are required to ensure the existence of the addon agent namespace(for example, add a namespace YAML file into manifests). 

And developers also should add the label "addon.open-cluster-management.io/namespace": "true" on the namespace, otherwise, addon-framework will not copy the image pull secret to the namespace and developers need to provide the image pull secret by themselves.